File: solutions/4 Median of Two Sorted Arrays [Hard]/sol.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        n1=len(nums1); n2=len(nums2)
        if n1>n2:
            return self.findMedianSortedArrays(nums2,nums1)
        n=n1+n2
        left=(n+1)//2
        low=0
        high=n1
        while low<=high:
            mid1 = (low+high) // 2
            mid2 = left-mid1
            l1,l2,r1,r2 = float("-inf"),float("-inf"),float("inf"),float("inf")
            if mid1<n1: r1=nums1[mid1]
            if mid2<n2: r2=nums2[mid2]
            if mid1>=1: l1=nums1[mid1-1]
            if mid2>=1: l2=nums2[mid2-1]
            if l1<=r2 and l2<=r1:
                if n%2==0: return (float(max(l1,l2)) + float(min(r1,r2))) / 2.0
                else: return max(l1,l2)
            if l1>r2: high=mid1-1
            else: low=mid1+1
        
        return 0
    
# Python    
# 78 ms (beats 71.85%)
# 17.04 mb (beats 6.51%)

File: solutions/9 Palindrome Number [Easy]/sol.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.size();
        int l = 0, r = 0;
        vector<int> chars(256, -1);
        int len = 0;
        while (r < n){
            if (chars[s[r]] != -1) l = max(chars[s[r]] + 1, l);
            chars[s[r]] = r;
            len = max(len, r - l + 1);
            r++;
        }

        return len;
    }
};

// C++
// 0 ms (beats 100.00%)
// 10.26 mb (beats 83.87%)